src/app/ReadHandler.h

```diff
@@ -247,6 +253,7 @@ class ReadHandler : public Messaging::ExchangeDelegate
         VerifyOrReturnError(aMaxInterval <= std::max(GetPublisherSelectedIntervalLimit(), mMaxInterval),
                             CHIP_ERROR_INVALID_ARGUMENT);
         mMaxInterval = aMaxInterval;
+        mObserver->OnReportingIntervalsChanged(this);
```

Comment:
i'm not sure if the observer should be notified here? This function is usually called during the establishment of a subscription...